openlibrary/plugins/upstream/borrow.py

```diff
@@ -138,8 +138,7 @@ def POST(self, key):
         edition = web.ctx.site.get(key)
         if not edition:
             raise web.notfound()
-        error_redirect = edition.url("/borrow")
-
+        error_redirect = "/account/login?redirect=%s/borrow" % edition.url()
```

Comment:
in core/models.py url() calls _get_url which calls _make_url which urlsafe's and urlencode's params:

https://github.com/internetarchive/openlibrary/blob/master/openlibrary/core/models.py#L127